src/uu/pidof/src/pidof.rs

```diff
@@ -0,0 +1,157 @@
+
+    let output = collected
+        .into_iter()
+        .map(|it| it.pid.to_string())
+        .collect::<Vec<_>>()
+        .join(arg_separator);
+
+    println!("{output}");
+
+    Ok(())
+}
+
+fn get_executable_name(process: &mut ProcessInformation) -> String {
+    let binding = process.cmdline.split(' ').collect::<Vec<_>>();
+    let mut path = binding.first().unwrap().to_string();
+
+    if path.is_empty() {
+        path.clone_from(&process.status()["Name"]);
+    };
+
+    PathBuf::from(path)
+        .file_name()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string()
+}
+
+fn collect_matched_pids(matches: &ArgMatches) -> Vec<ProcessInformation> {
+    let program_name: Vec<_> = matches
+        .get_many::<String>("program-name")
+        .unwrap()
+        .cloned()
+        .collect();
+
+    let collected = walk_process().collect::<Vec<_>>();
+    let arg_omit_pid = matches
+        .get_many::<usize>("o")
+        .unwrap_or_default()
+        .copied()
+        .collect::<Vec<_>>();
+
+    let mut processed = Vec::new();
+    for mut process in collected {
+        let contains = program_name.contains(&get_executable_name(&mut process));
+        let should_omit = arg_omit_pid.contains(&process.pid);
+
+        if contains && !should_omit {
+            processed.push(process)
+        }
+    }
+
+    let flag_s = matches.get_flag("s");
+    if flag_s {
+        match processed.first() {
+            Some(first) => vec![first.clone()],
+            None => Vec::new(),
+        }
+    } else {
+        processed
+    }
+}
+
+#[allow(clippy::cognitive_complexity)]
+pub fn uu_app() -> Command {
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .about(ABOUT)
+        .override_usage(format_usage(USAGE))
+        .infer_long_args(true)
+        .arg(
+            Arg::new("program-name")
+                .help("Program name.")
+                .required(true)
+                .index(1),
+        )
+        // .arg(
+        //     Arg::new("c")
+        //         .short('c')
+        //         .help("Return PIDs with the same root directory")
+        //         .action(ArgAction::SetTrue),
+        // )
+        .arg(
+            Arg::new("d")
+                .short('d')
+                .help("Use the provided character as output separator")
+                .action(ArgAction::Set)
+                .value_name("sep")
+                .default_value(" ")
+                .hide_default_value(true),
+        )
+        // .arg(
+        //     Arg::new("n")
+        //         .short('n')
+        //         .help("Avoid using stat system function on network shares")
+        //         .action(ArgAction::SetTrue),
+        // )
+        .arg(
+            Arg::new("o")
+                .short('o')
+                .help("Omit results with a given PID")
+                .action(ArgAction::Append)
+                .value_name("omitpid"),
+        )
```

Comment:
I think you have to specify a value parser for this option. Currently I get an error when using `-o`:
```
cargo run pidof -o1 init
thread 'main' panicked at src/uu/pidof/src/pidof.rs:66:10:
Mismatch between definition and access of `o`. Could not downcast to usize, need to downcast to alloc::string::String
```

Reply:
I'm sorry, my advice was incorrect :| The original `pidof` also allows multiple PIDs separated by `,` and so a value parser doesn't work.
```
# pidof sleep
40 39 38
# pidof -o40,39 sleep
38
```

Reply:
Humm I will take a look about it soon:P